### DISCUSSION TITLE
Allow clients to manually notify server when a scene is done loading.

### DISCUSSION THREAD

**Discussion by KeithSwanger** [FirstGearGames/FishNet#564] [Feature Request]
According to the SceneManager Scene Events diagram, loading a scene goes a little something like this:

1. The server loads a scene
2. The server tells a client to load the scene
3. The client loads the scene
4. **The client sends a "Load Completed" message to the server**
5. The server updates the visibility of observers for the client

It would be very useful if the **"Load Completed"** message sent from the client to the server could be delayed and manually sent later by the client. This would allow any asynchronous initialization of non-networked objects in the scene to be fully initialized client-side before observers are spawned for the client.

For a more concrete example, imagine I have a world generator that takes a seed and procedurally generates a terrain. This seed is synced in a Lobby scene. The server then tells the client to load the World scene.

When the client loads the World scene, the world generator uses the seed to generate the terrain, which will take some time to generate asynchronously. Since this generation takes time, **the client has already notified the server** that the World scene is loaded. The observers are spawned for the client, and they start glitching as they fall through an empty world with no terrain.

Right now, my solution has been to create a custom Observer Condition to assure that the client has fully initialized the scene. However, if we could manually decide when the **"Load Completed"** message is sent from the client to the server, this custom Observer Condition would be unnecessary!

**Comment by FirstGearGames** [comment]
This is already possible without any extra features needed. 

- Server tells client to load a scene (manually such as a RPC).
- Client loads the scenes, you track when it's completed locally using whatever means you like.
- Client sends RPC telling the server it's done.
- Server does SceneManager.AddConnectionToScene.

No custom observer conditions needed.

**Comment by KeithSwanger** [comment]
That seems very reasonable! Discussions on the discord never mentioned that technique, but that seems perfect for my use case. Thank you!
